app/models/service_template_provision_task.rb

```diff
@@ -158,6 +158,10 @@ def deliver_to_automate(req_type = request_type, _zone = nil)
     end
   end
 
+  def resource_action
+    source.resource_actions.detect { |ra| ra.action == 'Provision' } if source.respond_to?(:resource_actions)
```

Comment:
@bzwei This method name should be more descriptive.  My guess is you might be trying to abstract the "Provision" part away from the Transformation models.  If that is not correct this could be `provision_resource_action`.  

Otherwise we could use a more generic term like `initialize_resource_action` or `create_resource_action`.

If we abstract this method name it would allow the subclass to define a different `ra.action` value (currently "Provision") by moving the `"Provision"` value into a constant that the subclass could override.

Reply:
My understanding ruby's naming convention - `resource_action` means `get_resource_action`. I want to abstract the resource action, no matter where it come from. Can be from `source` or `miq_request`, or other place, can be type `Provision` or other type.

Reply:
I see that you are already overriding this method for the `ServiceTemplateTransformationPlanTask` class.

At the base class I feel it is misleading to have `resource_action` return the "Provision" action.  It would not be clear to me that `resource_action` == "Provision" but there are also "Reconfigure" and "Retirement" actions.  This is my main objection to the proposed method name.

Reply:
The base class is `ServiceTemplateProvisionTask`, so the action has to be `Provision`. A task can only have one action, for this task `Provision` is implied.

Reply:
Right, the piece I was overlooking was "A task can only have one action".  Thanks.